### PR TITLE
[ADD][9.0] account_analytic_distribution module

### PR DESCRIPTION
**Pull request by cubells**
New module to use as alternative to old account_analytic_plans module.

- Add spanish po file
- Add tests

cc @Tecnativa

### CODE REVIEW

File: account_analytic_distribution/README.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=============================
+Account Analytic Distribution
+=============================
+
+This module is a simpler alternative to official 'account_analytic_plans'
```
Comment (pedrobaeza) [inline account_analytic_distribution/README.rst]: Change this text, as now in version 9.0, there's no account_analytic_plans module, and also explains the purpose (as if you don't know account_analytic_plans).

File: account_analytic_distribution/README.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+addon provided by Odoo in version 8.0.
+
+Configuration
+=============
+
+To configure your analytic distributions go to
+Invoicing > Configuration > Analytic Accounting > Analytic Distributions
```
Comment (pedrobaeza) [inline account_analytic_distribution/README.rst]: Put a little more in this topic, specifying what fields you have to fill, and using `#. ` elements.

File: account_analytic_distribution/README.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+You can continue using 'Analityc Account' field, but 'Analytic Distribution'
+has precedence if defined.
+
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/87/9.0
+
+For further information, please visit:
```
Comment (pedrobaeza) [inline account_analytic_distribution/README.rst]: Remove this part

File: account_analytic_distribution/README.rst [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+* https://www.odoo.com/forum/help-1
+
+
+Known issues / Roadmap
+======================
+
+* Analytic distribution can only be defined at Journal Entry form.
+TODO: Add 'Analytic Distribution' field to invoices and other account objects
```
Comment (pedrobaeza) [inline account_analytic_distribution/README.rst]: Add this as a bullet point

File: account_analytic_distribution/README.rst [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+There will be developed other addons (like 'sale_analytic_distribution' or
```
Comment (pedrobaeza) [inline account_analytic_distribution/README.rst]: Add this as a bullet point

File: account_analytic_distribution/models/__init__.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Antonio Espinosa - <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+# flake8: noqa
```
Comment (pedrobaeza) [inline account_analytic_distribution/models/__init__.py]: Remove this line as this a hack for not checking flake8, and fix possible errors.

File: account_analytic_distribution/models/account_analytic_distribution.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    )
+    rule_ids = fields.One2many(
+        string="Distribution rules",
+        comodel_name='account.analytic.distribution.rule',
+        inverse_name='distribution_id',
+    )
+
+    _sql_constraints = [
+        ('name_uniq', 'unique(name, company_id)',
+            _('Distribution name must be unique per Company!')),
```
Comment (pedrobaeza) [inline account_analytic_distribution/models/account_analytic_distribution.py]: Fix indentation

File: account_analytic_distribution/models/account_analytic_distribution.py [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    ]
+
+    @api.model
+    def get_default_company(self):
```
Comment (pedrobaeza) [inline account_analytic_distribution/models/account_analytic_distribution.py]: Why 2 methods for the same? I supposed that this is due to old API not letting to overwrite, but now you can make only one

File: account_analytic_distribution/models/account_analytic_distribution_rule.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Antonio Espinosa - <antonio.espinosa@tecnativa.com>
+# Copyright 2017 Vicent Cubells - <vicent.cubells@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import _, fields, models
+
+
+class AccountAnalyticDistributionRule(models.Model):
```
Comment (pedrobaeza) [inline account_analytic_distribution/models/account_analytic_distribution_rule.py]: This should go in the same file as the account.analytic.distribution, as it's a one2many model (like sale.order.line).

File: account_analytic_distribution/README.rst (hunk 4 above)
Comment (pedrobaeza) [reply account_analytic_distribution/README.rst]: Is this still TODO? I see the distribution in the invoice line...

File: account_analytic_distribution/tests/test_analytic_distribution.py [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+from openerp.tests import common
+from openerp.exceptions import ValidationError
+
+
+class TestAnalyticDistribution(common.SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestAnalyticDistribution, cls).setUpClass()
+        cls.account1 = cls.env.ref('analytic.analytic_agrolait')
```
Comment (pedrobaeza) [inline account_analytic_distribution/tests/test_analytic_distribution.py]: Don't use demo data, but create yours

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
@cubells please implement the policy per company as discussed.

**Comment by cubells** [comment]
@pedrobaeza done.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@chienandalu @luismontalba please review

**Comment by fclementic2c** [comment]
@cubells @pedrobaeza I wanted to test this tool on runbot but I can't make it works :\ 
look at the invoice : INV/2017/0005, I do not get 2 analytic lines per invoice line as expected....maybe I did something wrong.

**Comment by cubells** [comment]
@fclementic2c 
I'll test this problem...

**Comment by cubells** [comment]
@fclementic2c 
The problem was that in runbot module account_analytic_no_lines was installed.

As stated in its README file:

> This module hides analytics lines from accounting menus and disable their generation from an invoice or a move line.

Can you retest this module with account_analytic_no_lines not installed?

**Review by luismontalba — APPROVED** [review]
Functionality tested on runbot

**Comment by fclementic2c** [comment]
@cubells 
Its works better thanks :)

I just have one little concern about ergonomy: why not adding a constraint on so/po/invoices lines: if you select a distribution then block the analytic account field (since this info will be overwritten anyway on analytic lines)

Otherwise, it is ok for me
Thanks for your module

:+1:

**Pull request merged**